### PR TITLE
Nutrient Levels Templatization

### PR DESCRIPTION
**Pull request by areeshatariq**
**Description:**
HTML Templatization of Nutrient levels using Template::Toolkit

**Related issues and discussion:** #2416

### CODE REVIEW

File: lib/ProductOpener/Display.pm [hunk 1]
```diff
@@ -8842,9 +8848,12 @@ sub display_nutrient_levels($) {
 
 	my $html_nutrition_grade = '';
 	my $html_nutrient_levels = '';
-
+
```
Comment (Ban3) [inline lib/ProductOpener/Display.pm:8851]:
```suggestion

```

File: lib/ProductOpener/Display.pm [hunk 2]
```diff
@@ -8800,6 +8800,12 @@ sub display_nutrient_levels($) {
 
 	my $html = '';
 
+	my $template_data_ref = {
+		
+		lang => \&lang,
```
Comment (Ban3) [inline lib/ProductOpener/Display.pm:8805]:
```suggestion

		lang => \&lang,
```

File: lib/ProductOpener/Display.pm [hunk 3]
```diff
@@ -8858,6 +8867,7 @@ sub display_nutrient_levels($) {
 				and (defined $product_ref->{nutrition_score_warning_no_fruits_vegetables_nuts})
 					and ($product_ref->{nutrition_score_warning_no_fruits_vegetables_nuts} == 1)) {
 				$warning .= "<p>" . lang("nutrition_grade_fr_fiber_and_fruits_vegetables_nuts_warning") . "</p>";
+
```
Comment (Ban3) [inline lib/ProductOpener/Display.pm:8870]:
```suggestion

```

File: lib/ProductOpener/Display.pm [hunk 4]
```diff
@@ -8885,7 +8895,7 @@ sub display_nutrient_levels($) {
 			}
 		}
 
-		$html_nutrition_grade .= <<HTML
+ 		$html_nutrition_grade .= <<HTML
```
Comment (Ban3) [inline lib/ProductOpener/Display.pm:8898]:
```suggestion
		$html_nutrition_grade .= <<HTML
```

File: templates/nutrient_levels.tt.html [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        </div>
+    [% END %]
+
+[% ELSIF (nutrition_grade != "exists") && (nutrient_levels != '') %]
+    <h4>[% lang('nutrient_levels_info') %]
+        <a href="[% lang('nutrient_levels_link') %]" title="[% lang('nutrient_levels_info') %]">
+            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" class="icon" aria-hidden="true" focusable="false"><path d="M0 0h24v24H0z" fill="none"/><path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"/></svg>
+        </a>
+    </h4>
+    [% FOREACH level IN nutrient_levels %]
```
Comment (Ban3) [inline templates/nutrient_levels.tt.html:23]:
```suggestion
    [% FOREACH level IN nutrient_levels %]
```

File: templates/nutrient_levels.tt.html [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<!-- Nutrient levels template -->
+[% IF (nutrition_grade == "exists") && (nutrient_levels != '') %]
+    [% IF nutrient_levels != '' %]
+        <div class="small-12 xlarge-6 columns">
+            <h4>[% lang('nutrient_levels_info') %]
+                <a href="[% lang('nutrient_levels_link') %]" title="[% lang('nutrient_levels_info') %]">
+                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" class="icon" aria-hidden="true" focusable="false"><path d="M0 0h24v24H0z" fill="none"/><path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"/></svg>
+                </a>
+            </h4>
+            [% FOREACH level IN nutrient_levels %]
```
Comment (Ban3) [inline templates/nutrient_levels.tt.html:10]:
```suggestion
            [% FOREACH level IN nutrient_levels %]
```

File: templates/nutrient_levels.tt.html [hunk 7]
```diff
@@ -0,0 +1,28 @@
+<!-- Nutrient levels template -->
+[% IF (nutrition_grade == "exists") && (nutrient_levels != '') %]
+    [% IF nutrient_levels != '' %]
+        <div class="small-12 xlarge-6 columns">
+            <h4>[% lang('nutrient_levels_info') %]
+                <a href="[% lang('nutrient_levels_link') %]" title="[% lang('nutrient_levels_info') %]">
+                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" class="icon" aria-hidden="true" focusable="false"><path d="M0 0h24v24H0z" fill="none"/><path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"/></svg>
```
Comment (stephanegigandet) [inline templates/nutrient_levels.tt.html:7]: I don't think it's a good idea to put the SVG in there, we should pass a parameter to the template with the output of display_icon()

Comment (stephanegigandet) [reply templates/nutrient_levels.tt.html:7]: Or maybe just call display_icon() directly in the template

File: templates/nutrient_levels.tt.html [hunk 8]
```diff
@@ -0,0 +1,28 @@
+<!-- Nutrient levels template -->
+[% IF (nutrition_grade == "exists") && (nutrient_levels != '') %]
```
Comment (stephanegigandet) [inline templates/nutrient_levels.tt.html:2]: With this if statement, you have to repeat the template twice. I think we should remove the if statement, especially as it is still in the Perl code.

What it means is that this template should display only the nutrient levels, but not the "<div class="small-12 xlarge-6 columns">" that may be added (but not always) before it.

**Pull request merged**